pandas/tests/scripts/test_validate_docstrings.py

```diff
@@ -193,6 +193,25 @@ def contains(self, pat, case=True, na=np.nan):
         """
         pass
 
+    def mode(self, axis, numeric_only):
+        """
+        Ensure sphinx directives don't affect checks for trailing periods.
+
+        Parameters
+        ----------
+        axis : str
+            Sentence ending in period, followed by single directive.
+
+            .. versionchanged:: 0.1.2
+
+        numeric_only : boolean
+            Sentence ending in period, followed by multiple directives.
+
+            .. versionadded:: 0.1.2
+            .. deprecated:: 0.00.0
```

Comment:
I'm not sure about `versionadded` and `versionchanged`, but `deprecated` can have a description after if, for example:

```
          .. deprecated:: 0.21.0
              Use :func:`pandas.read_csv` instead.
```

And it can be even multiline. Do you mind adding a test for that? I'm not sure if this is working with the current implementation.

Reply:
Also, if you check the `convert_datetime64` of `to_records`, there are cases where the directives come before the description. I'm happy if we consider only valid having them in one place (before or after the description). But, can we make the script generate a descriptive error for it? I guess with the current implementation we'll report that the parameter has no description.